examples/hal/quad/main.rs

```diff
@@ -63,6 +63,10 @@ const COLOR_RANGE: i::SubresourceRange = i::SubresourceRange {
 #[cfg(any(feature = "vulkan", feature = "dx12", feature = "metal", feature = "gl"))]
 fn main() {
     env_logger::init().unwrap();
+
+    #[cfg(feature = "metal")]
+    let mut autorelease_pool = unsafe { back::AutoreleasePool::new() };
```

Comment:
I believe `hal/compute` and `warden` also need to be updated?
I wonder if we could make the `AutoreleasePool` a bit more convenient... say, do `release` in `drop()` for example

Reply:
It does release in `drop`, but since releasing the AutoreleasePool at the wrong time can introduce memory unsafety I opted for calling an unsafe `reset` method on the same `back::AutoreleasePool` (which swaps the `NSAutoreleasePool` internally) instead of relying on an implicit drop to determine when that gets called. The unsafety in the `Drop` implementation is the reason I made `AutoreleasePool::new` unsafe, so that at least  you can't shoot yourself in the foot without including *some* related unsafe code.